src/content/get-started/fundamentals/dart.md

```diff
@@ -0,0 +1,179 @@
+to anyone who's written Java, Javascript,
+or any other C-like language.  
+
+:::note
+Installing Flutter also installs Dart,
+so you don't need to install Dart separately.
+:::
+
+The following example is a small program that 
+fetches data from dart.dev, 
+decodes the returned json, 
+and prints it to the console. 
+If you’re confident in your ability to 
+understand this program, 
+feel free to skip to the page.
+
+```dart
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+
+class Package {
+  final String name, latestVersion; 
+  String? description;
+
+  Package(this.name, this.latestVersion, this.description);
+
+  @override
+  String toString() {
+    return 'Package{name: $name, latestVersion: $latestVersion, description: $description}';
+  }
+}
+
+void main() async {
+  final httpPackageUrl = Uri.https('dart.dev', '/f/packages/http.json');
+  final httpPackageResponse = await http.get(httpPackageUrl);
+  if (httpPackageResponse.statusCode != 200) {
+    print('Failed to retrieve the http package!');
+    return;
+  }
+  final json = jsonDecode(httpPackageResponse.body);
+  final package = Package(json['name'], json['latestVersion'], json['description']);
+  print(package);
+}
+```
+
+This program has two parts: 
+the `Package` class declaration, and the business logic, 
+which is contained in the [`main`][] function.
+
+The `Package` class contains many of the most common
+features you’ll use when working with [classes in Dart][].
+This class has three members, 
+and defines a constructor and a method.
+
+The Dart language is [type safe][]; it uses 
+static type checking to ensure that 
+a variable's value always matches the
+variable's static type. 
+When defining a class, annotating the members with 
+`String` is required, 
+but it is often optional due to type inference. 
+In the `main` function in this example 
+there are many lines that start with `final variableName =`. 
+These lines are type safe, 
+despite not being explicitly given a type.
+
+Dart also has built-in [sound null safety][]. 
+In the example, the `description` member is 
+declared with the type `String?`. 
+The `?` at the end of `String?` means that 
+this property can be null. 
+The other two members cannot be null, 
+and the program will not compile if 
+you tried to set them to null. 
+You can see this demonstrated in the constructor for 
+the ‘Package class’. It takes two required,
+positional arguments and one optional, named argument.
+
+Next in the example is the `main` function. 
+All Dart programs, including Flutter apps, 
+start with a `main`function. 
+The function showcases several basic Dart language features, 
+including using libraries, marking functions as async, 
+making function calls, using `if` statement control-flow,
+and more.
+
+:::note Where does initialization code go?
+The main entrypoint in a starter
+Flutter app is in `lib/main.dart`.
+The default `main` method looks
+like the following:
+
+```dart
```

Comment:
Consider adding the path to the code block since it seems relevant:

````suggestion
```dart title="lib/main.dart"
````